ofctrl/fgraphGroup.go

```diff
@@ -109,13 +109,24 @@ func (self *Group) getGroupModMessage(command int) *openflow15.GroupMod {
 		groupMod.Type = openflow15.GT_FF
 	}
 
-	for _, bkt := range self.Buckets {
-		// Add the bucket to group
-		groupMod.AddBucket(*bkt)
+	// In a request of type OFPGC_REMOVE_BUCKET, bucket list should be empty.
+	if command != openflow15.OFPGC_REMOVE_BUCKET {
+		for _, bkt := range self.Buckets {
+			// Add the bucket to group
+			groupMod.AddBucket(*bkt)
+		}
 	}
-
 	if command == openflow15.OFPGC_INSERT_BUCKET {
+		// In a request of type OFPGC_INSERT_BUCKET, the command_bucket_id field is used to specify the position in the
+		// current action bucket list to insert new action buckets. Here place the buckets to the end of the current action
+		// bucket list.
 		groupMod.CommandBucketId = openflow15.OFPG_BUCKET_LAST
+	} else if command == openflow15.OFPGC_REMOVE_BUCKET {
+		// In a request of type OFPGC_REMOVE_BUCKET, the CommandBucketId field is used to specify the identifier of the
+		// bucket to remove from current action bucket list.
```

Comment:
What does "current action bucket list" mean, the buckets in the groupMod message or the buckets that already realized on the OF Switch ( OVS )?

Reply:
It means that the buckets have been installed.

Reply:
Maybe update the comment?

Reply:
Updated